Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java

```diff
@@ -473,52 +502,359 @@ public BlackboardArtifact getArtifact() {
         return this.artifact;
     }
 
+    /**
+     * Returns a list of non null actions from the given possibly null options.
+     *
+     * @param items The items to purge of null items.
+     *
+     * @return The list of non-null actions.
+     */
+    private List<Action> getNonNull(Action... items) {
+        return Stream.of(items)
+                .filter(i -> i != null)
+                .collect(Collectors.toList());
+    }
+
     @Override
     public Action[] getActions(boolean context) {
-        List<Action> actionsList = new ArrayList<>();
-        actionsList.addAll(Arrays.asList(super.getActions(context)));
+        // groupings of actions where each group will be separated by a divider
+        List<List<Action>> actionsLists = new ArrayList<>();
+
+        // view artifact in timeline
+        actionsLists.add(getNonNull(
+                getTimelineArtifactAction(this.artifact)
+        ));
+
+        // view associated file (TSK_PATH_ID attr) in directory and timeline
+        actionsLists.add(getAssociatedFileActions(this.artifact, this.artifactType));
+        
+        // view source content in directory and timeline
+        actionsLists.add(getNonNull(
+                getViewSrcContentAction(this.artifact, this.srcContent),
+                getTimelineSrcContentAction(this.srcContent)
+        ));
+
+        // extract with password from encrypted file
+        actionsLists.add(getNonNull(
+                getExtractWithPasswordAction(this.srcContent)
```

Comment:
Ultimately, yes, but not within the scope of the story behind this PR. We are in the process of looking at a general solution (e.g., perhaps the AutopsyDataModeItem class hierarchy) and I think that we should decide what the general solution is rather than fix these issues piecemeal).